sgit/core.py

```diff
@@ -576,24 +623,30 @@ def _select(self, sequence, selection_query, selection_method):
             elif selection_query == "first":
                 return sequence[0]
             else:
-                spec = SpecifierSet(selection_query)
+                log.debug(f"Selection query")
 
-                filtered_versions = list(
-                    spec.filter(
-                        sequence,
+                try:
+                    spec = SpecifierSet(selection_query)
+
+                    filtered_versions = list(
+                        spec.filter([str(item) for item in sequence]),
                     )
-                )
 
-                log.debug(f"filtered_versions")
-                log.debug(filtered_versions)
+                    log.debug(f"filtered_versions")
+                    log.debug(filtered_versions)
 
-                return filtered_versions[-1]
-        elif selection_method == SelectionMethods.EXACT:
+                    return filtered_versions[-1]
+                except packaging.specifiers.InvalidSpecifier:
+                    log.warning(f"WARNING: Invalid SEMVER select query. Falling back to EXCAT matching of value")
+                    selection_method = SelectionMethods.EXACT
+        
+        if selection_method == SelectionMethods.EXACT:
```

Comment:
```suggestion
        elif selection_method == SelectionMethods.EXACT:
```

Reply:
This will fail out and this will break the overriding selection method. If you look two lines up at L641 you will see that i set EXACT method as a fallback if SEMVER fails out, so if we change to elif EXACT fallback method will never be reached